keps/sig-apps/0753-sidecarcontainers.md

```diff
@@ -109,34 +180,173 @@ section][#graduation-criteria].
+can sometimes force the coupling of application code with infrastructure. This
+causes a dependency between applications and infrastructure, and forces more
+coordination between multiple, possibly independent, teams.
+
+An example in the open source world about this is the [Istio CNI
+plugin][istio-cni-plugin]. This was created as an alternative for the
+initContainer hack that the service mesh needs to do. The initContainer will
+blackhole all traffic until the Istio container is up. The CNI plugin can be
+used to completely remove the need for such an initContainer. But this
+alternative requires that nodes have the CNI plugin installed, effectively
+coupling the service mesh app with the infrastructure.
+
+This KEP removes the need for a service mesh to use either an initContainer or a
+CNI plugin: just guarantee that the sidecar container can be started first.
+
+While in this specific example the CNI plugin has some benefits too (removes the
+need for some capabilities in the pod) and might be worth pursuing, it is used
+here as an example to show thee possibility of coupling apps with
+infrastructure. Similar examples also exist in-house for non-open source
+applications too.
+
+[istio-cni-plugin]: https://istio.io/latest/docs/setup/additional-setup/cni/
 
 ## Goals
 
+This proposal aims to:
+ * Allow Kubernetes Jobs to have sidecar containers that run continuously
+   without any coupling between containers in the pod.
+ * Allow pods to start a subset of containers first and, only when those are
+   ready, start the remaining containers.
+ * Allow pods to stop a subset of containers first and, only when those have
+   been stopped, stop the rest.
+ * Not change the semantics in any way for pods not using this feature.
+ * Only change the semantics of startup/shutdown behaviour for pods using this
+   feature.
+
 Solve issues so that they don't require application modification:
 * [25908](https://github.com/kubernetes/kubernetes/issues/25908) - Job completion
-* [65502](https://github.com/kubernetes/kubernetes/issues/65502) - Container startup dependencies
+* [65502](https://github.com/kubernetes/kubernetes/issues/65502) - Container startup dependencies. Istio bug report
 
 ## Non-Goals
 
+This proposal doesn't aim to:
+ * Allow a multi-level ordering of containers start/stop sequence. Only two
+   types are defined: "sidecar" or "standard" containers. It is not a goal to
+   have more than these two types.
+ * Model startup/shutdown dependencies between different sidecar containers
+ * Change the pod phase in any way, with or without this feature enabled
+ * Reimplement an init-system alike semantics for pod containers
+   startup/shutdown
+ * Allow sidecar containers to run concurrently with initContainers
+
 Allowing multiple containers to run at once during the init phase - this could be solved using the same principal but can be implemented separately. //TODO write up how we could solve the init problem with this proposal
```

Comment:
Is this a duplication of an item above `Allow sidecar containers to run concurrently with initContainers`?

Reply:
Yes, I just left it as it had a "TODO" note. I can remove it :)